Source/WebKit/UIProcess/API/C/WKPage.cpp

```diff
@@ -3042,23 +3042,23 @@ void WKPageDidDenyPointerLock(WKPageRef pageRef)
 {
     CRASH_IF_SUSPENDED;
 #if ENABLE(POINTER_LOCK)
-    toImpl(pageRef)->didDenyPointerLock();
+    toProtectedImpl(pageRef)->didDenyPointerLock();
 #else
     UNUSED_PARAM(pageRef);
 #endif
 }
 
 void WKPagePostMessageToInjectedBundle(WKPageRef pageRef, WKStringRef messageNameRef, WKTypeRef messageBodyRef)
 {
-    toImpl(pageRef)->postMessageToInjectedBundle(toImpl(messageNameRef)->string(), toImpl(messageBodyRef));
+    toProtectedImpl(pageRef)->postMessageToInjectedBundle(toProtectedImpl(messageNameRef)->string(), toImpl(messageBodyRef));
 }
 
 WKArrayRef WKPageCopyRelatedPages(WKPageRef pageRef)
 {
     Vector<RefPtr<API::Object>> relatedPages;
 
-    for (Ref page : toImpl(pageRef)->protectedLegacyMainFrameProcess()->pages()) {
-        if (page.ptr() != toImpl(pageRef))
+    for (Ref page : toProtectedImpl(pageRef)->protectedLegacyMainFrameProcess()->pages()) {
```

Comment:
It seems we should cache in a variable to avoid repeated calls to `toProtectedImpl(pageRef)` here and inside the loop.